pandevice/base.py

```diff
@@ -4247,10 +4252,22 @@ def _commit(self, cmd=None, exclude=None, commit_all=False,
             pass
         else:
             cmd = ET.Element("commit")
-            if exclude is not None:
-                excluded = ET.SubElement(cmd, "partial")
-                excluded = ET.SubElement(excluded, exclude)
-            cmd = ET.tostring(cmd, encoding='utf-8')
+            if exclude is not None or admin is not None:
+                if admin is not None:
+                    excluded = ET.SubElement(cmd, "partial")
+                    excluded = ET.SubElement(excluded, "admin")
+                    if isinstance(admin, str):
+                        admin = [admin]
+                    admin_length = len(admin)
+                    for i in range(admin_length):
+                        admin_xml = ET.SubElement (excluded, "member")
+                        admin_xml.text = admin[i]
+                    cmd = ET.tostring(cmd, encoding='utf-8')
+                if exclude is not None:
+                    excluded = ET.SubElement(cmd, "partial")
```

Comment:
You are adding a second "partial" block if you do this, which is not correct.  So beneath the new line 4255, you need to add the "partial" subelement only once, then both the exclude code and the admin code should use that as the attach point.

Reply:
@shinmog - See requested updates for all three comments in the new commit.